version/distro/distro.go

```diff
@@ -94,3 +95,23 @@ func freebsdDistro() Distro {
 	}
 	return ""
 }
+
+// DSMVersion reports the Synology DSM major version.
+//
+// If not Synology, it reports 0.
+func DSMVersion() int {
+	if runtime.GOOS != "linux" {
+		return 0
+	}
+	if Get() != Synology {
+		return 0
+	}
+	v, _ := os.ReadFile("/etc/VERSION")
+	if bytes.Contains(v, []byte(`majorversion="6"`)) {
```

Comment:
There was at least one person who managed to get DSM5.1 sortof working. I think they had to disable something about certificate checking, as the CA roots were so old.

Reply:
I don't want to start supporting DSM5. We never have, we don't have test devices, we rarely hear about it, etc.